src/controller/CHIPDevice.h

```diff
@@ -351,6 +357,23 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     ByteSpan GetCSRNonce() const { return ByteSpan(mCSRNonce, sizeof(mCSRNonce)); }
 
+    /*
+     * This function can be called to establish a secure session with the device.
+     *
+     * If the device doesn't have operational credentials, and is under commissioning process,
+     * PASE keys will be used for secure session.
+     *
+     * If the device has been commissioned and has operational credentials, CASE session
+     * setup will be triggered.
+     *
+     * On establishing the session, the callback function `onConnection` will be called. If the
+     * session setup fails, `onFailure` will be called.
```

Comment:
... These callbacks must not be already registered with a source (i.e., if they were used previously, they must be canceled).

And add a precondition check to the implementation to verify that.